airbyte-ci/connectors/pipelines/pipelines/cli/airbyte_ci.py

```diff
@@ -56,7 +57,7 @@ def display_welcome_message() -> None:
     )
 
 
-def check_up_to_date() -> bool:
+def check_up_to_date(throw_as_error=False) -> bool:
```

Comment:
nit:
I'd suggest making this function always succeed (returns true or false)
And implement the raising logic at the caller level.
It'd be better to throw a `ClickException` as it has a `show` method to cleanly output errors to `stderr`.
I think taking this approach would help build more environment checks in a consolidated way.